frontend/src/pages/UsersPage.jsx

```diff
@@ -1,10 +1,98 @@
+import React, { useEffect, useState, useCallback } from 'react';
+import { useParams, useNavigate } from 'react-router';
+import { useUser } from '../hooks/useUser';
+import { useApi } from '../hooks/useApi';
+import Spinny from '../components/Spinny';
+import { Pagination } from '../components/Pagination';
+import { Link } from 'react-router';
+
 const UsersPage = () => {
+  const { userState, isLogged } = useUser();
+  const navigate = useNavigate();
+  const { page = 1 } = useParams();
+  const currentPage = parseInt(page, 10) - 1;
+
+  const [users, setUsers] = useState([]);
+  const [totalPages, setTotalPages] = useState(0);
+
+  const { data, error, isLoading, fetchFromApi } = useApi({ method: 'GET' });
+
+  const count = 20;
+
+  const getApiUrl = useCallback(
+    (pageNo = 0) => {
+      const skip = pageNo > 0 ? `&skip=${pageNo * count}` : '';
+      return `/users?limit=${count}${skip}`;
+    },
+    [count]
+  );
+
+  const getPageUrl = useCallback(pageNo => `/users/page/${pageNo + 1}`, []);
+
+  useEffect(() => {
+    if (!isLogged) {
+      navigate('/login');
+      return;
+    }
+    if (!userState.is_admin) {
+      navigate('/');
+      return;
+    }
+
+    fetchFromApi(getApiUrl(currentPage));
+  }, [
+    isLogged,
+    userState.is_admin,
+    navigate,
+    currentPage,
+    fetchFromApi,
+    getApiUrl,
+  ]);
+
+  useEffect(() => {
+    if (data && !error) {
+      setUsers(data.users);
+      const pages = Math.ceil(data.count / count);
+      setTotalPages(pages);
+    }
+    if (error) {
+      console.error('Error fetching users:', error);
+    }
+  }, [data, error, count]);
+
+  if (isLoading) {
```

Comment:
There's interesting bug with how `isLoading` is changed, when pagination triggers fetching another page.

To reproduce:
1. Change `count`, so there's pagination.
2. Play around with going to different pages. Notice the _active page_ button doesn't change, while everything else appears to work correctly - changing displayed users, changing URL.
3. When refreshed, or directly entering `/page/x`, the _active page_ button is correct.

I'm explaining it to myself the following way:
- Clicking on another page button triggers fetching and changes `isLoading` to `true`.
- Everything is replaced by `Spinny`.
- When fetch is finished `Pagination` component is rendered again.
- However the `currentPage` internal of the `UsersPage` component - does not change to reflect the changed URL, or different users displayed.
- Therefore `Pagination` component renders again with the original `currentPage`, what results in wrong _active page_.

```suggestion
  if (isLoading && users.length === 0) {
```